library/java/src/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -1,11 +1,50 @@
 package io.envoyproxy.envoymobile.engine;
 
-public interface EnvoyConfiguration {
+public class EnvoyConfiguration {
+
+  public final Integer connectTimeoutSeconds;
+  public final Integer dnsRefreshSeconds;
+  public final Integer statsFlushSeconds;
 
   /**
-   * Provides a default configuration template that may be used for starting Envoy.
+   * Create an EnvoyConfiguration with a user provided configuration values.
    *
-   * @return A template that may be used as a starting point for constructing configurations.
+   * @param connectTimeoutSeconds timeout for new network connections to hosts in the cluster.
+   * @param dnsRefreshSeconds rate in seconds to refresh DNS.
+   * @param statsFlushSeconds interval at which to flush Envoy stats.
    */
-  String templateString();
+  public EnvoyConfiguration(int connectTimeoutSeconds, int dnsRefreshSeconds,
+                            int statsFlushSeconds) {
+    this.connectTimeoutSeconds = connectTimeoutSeconds;
+    this.dnsRefreshSeconds = dnsRefreshSeconds;
+    this.statsFlushSeconds = statsFlushSeconds;
+  }
+
+  /**
+   * Resolves the provided default configuration with the field values:
```

Comment:
I'd probably remove the list of values below and simply say `Resolves the provided configuration template using properties on this configuration.`, that way we don't need to update this docstring each time we add a config property